### PR TITLE
Implement metric for remaining KES periods

### LINKED ISSUE #1030: CLI - Metric for remaining KES periods
We should have a metric for the remaining time, and remaining number of KES periods before the hot key's operational certificate expires.

### PR DESCRIPTION
**Pull request by intricate**
Closes #1030

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 1]
```diff
@@ -665,6 +671,65 @@ mpTracer :: ( Show (ApplyTxErr blk)
...
+            -- The KES period of the hot key is relative to the start KES
+            -- period of the operational certificate.
+            currentKesPeriod = oCertStartKesPeriod + kesPeriodOfKey
+
+            oCertExpiryKesPeriod = oCertStartKesPeriod + fromIntegral maxKesEvos
+
+            -- TODO @intricate: What if
+            --                  @currentKesPeriod > oCertExpiryKesPeriod@?
+            kesPeriodsUntilExpiry =
+              max 0 (oCertExpiryKesPeriod - currentKesPeriod)
```
Comment (intricate) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:701]: Forgot to drop this `TODO` comment.

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 2]
```diff
@@ -665,6 +670,63 @@ mpTracer :: ( Show (ApplyTxErr blk)
...
+
+            logValue :: LOContent a
+            logValue =
+              LogValue "remainingKESPeriods"
+                $ PureI
+                $ fromIntegral kesPeriodsUntilExpiry
+
+        meta <- mkLOMeta Critical Confidential
+        traceNamedObject tr' (meta, logValue)
```
Comment (dcoutts) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:707]: Do we not also want to trace the other values? The `oCertStartKesPeriod`, `oCertExpiryKesPeriod` and `currentKesPeriod`? Might as well.

Comment (disassembler) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:707]: yes please :)

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 3]
```diff
@@ -665,6 +670,73 @@ mpTracer :: ( Show (ApplyTxErr blk)
...
+                  $ fromIntegral oCertStartKesPeriod
+              , LogValue "operationalCertificateExpiryKESPeriod"
+                  $ PureI
+                  $ fromIntegral oCertExpiryKesPeriod
+              , LogValue "currentKESPeriod"
+                  $ PureI
+                  $ fromIntegral currentKesPeriod
+              , LogValue "remainingKESPeriods"
+                  $ PureI
+                  $ fromIntegral kesPeriodsUntilExpiry
```
Comment (dcoutts) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:713]: Nice :+1:

**Pull request merged**